Tests/NIOTests/BootstrapTest.swift

```diff
@@ -529,6 +529,309 @@ class BootstrapTest: XCTestCase {
+                    .serverOptions([.allowImmediateEndpointAddressReuse])
+                    .childChannelInitializer { channel in
+                        optionRead = channel.getOption(option)
+                        serverAcceptedChannelPromise.succeed(channel)
+                        return channel.pipeline.addHandler(CloseHandler())
+                    }.bind(host: "127.0.0.1", port: 0).wait())
+
+                let clientChannel = try assertNoThrowWithValue(clientBootstrap
+                    .channelInitializer({ (channel: Channel) in channel.eventLoop.makeSucceededFuture(()) })
+                    .connect(host: "127.0.0.1", port: serverChannel.localAddress!.port!).wait())
+
+                let serverAcceptedChannel = try serverAcceptedChannelPromise.futureResult.wait()
+
+                // Start shutting stuff down.
+                XCTAssertNoThrow(try clientChannel.close().wait())
+
+                // Wait for the close promises. These fire last.
+                XCTAssertNoThrow(try EventLoopFuture.andAllSucceed([clientChannel.closeFuture,
+                                                                    serverAcceptedChannel.closeFuture],
+                                                                    on: group.next()).wait())
+            XCTAssertNotNil(optionRead)
+            return try optionRead!.wait()
+
+        }
+        
+        func checkOptionEquivalence<Option>(longOption: Option, setValue: Option.Value,
+                                            shortOption: NIOTCPShorthandOption) throws
+            where Option : ChannelOption, Option.Value : Equatable {
+            
+            let longSetValue = try setAndGetOption(option: longOption) { bs in
+                bs.childChannelOption(longOption, value: setValue)
+            }
+            let shortSetValue = try setAndGetOption(option: longOption) { bs in
+                bs.childChannelOptions([shortOption])
+            }
+            let unsetValue = try setAndGetOption(option: longOption) { $0 }
+            
+            XCTAssertEqual(longSetValue, shortSetValue)
+            XCTAssertNotEqual(longSetValue, unsetValue)
+        }
+        
+        // At least on Darwin the default for child is to have allow reuse set - probably inherited from listen
+        // try checkOptionEquivalence(longOption: ChannelOptions.socketOption(.reuseaddr),
+        //                           setValue: 1,
+        //                           shortOption: .allowImmediateEndpointAddressReuse)
+        try checkOptionEquivalence(longOption: ChannelOptions.allowRemoteHalfClosure,
+                                   setValue: true,
+                                   shortOption: .allowRemoteHalfClosure)
+        // try checkOptionEquivalence(longOption: ChannelOptions.autoRead,
+           //                        setValue: false,
+             //                      shortOption: .disableAutoRead)
```

Comment:
There's a lot of commenting out here, what's that for?

Reply:
It's sort of an acknowledgment that my test doesn't work properly - gets stuck with auto-read off.

Reply:
Fixed the second commented test, kept the comment but not the code for the first.